pandas/tests/indexing/test_loc.py

```diff
@@ -773,3 +773,35 @@ def test_series_indexing_zerodim_np_array(self):
         s = Series([1, 2])
         result = s.loc[np.array(0)]
         assert result == 1
+
+    def test_PeriodIndex_str_slicing(self):
+        ix = pd.period_range(start='2017-01-01', end='2018-01-01', freq='M')
+        ser = ix.to_series()
+        result = ser.loc[:"2017-12"]
+        expected = ser.iloc[:-1]
+
+        tm.assert_series_equal(result, expected)
+
+    def test_TimedeltaIndex_str_slicing(self):
+        ix = pd.timedelta_range(start='1 day', end='2 days', freq='1H')
+        ser = ix.to_series()
+        result = ser.loc[:"1 days"]
+        expected = ser.iloc[:-1]
+
+        tm.assert_series_equal(result, expected)
+
+    def test_PeriodIndex_slicing(self):
```

Comment:
these could go in pandas/tests/indexing/test_datetimes.py or if you want to create a test_period.py would be ok too.

Reply:
Done.